network_provisioning/CMakeLists.txt

```diff
@@ -0,0 +1,33 @@
+if (NOT DEFINED ENV{IDF_PATH})
+    message(FATAL_ERROR "Please set IDF_PATH to the path of esp-idf repo")
+endif(NOT DEFINED ENV{IDF_PATH})
+
+set(IDF_PATH $ENV{IDF_PATH})
+
+set(srcs "src/network_config.c"
+        "src/network_scan.c"
+        "src/network_ctrl.c"
+        "src/manager.c"
+        "src/handlers.c"
+        "src/scheme_console.c"
+        "proto-c/network_config.pb-c.c"
+        "proto-c/network_scan.pb-c.c"
+        "proto-c/network_ctrl.pb-c.c"
+        "proto-c/network_constants.pb-c.c")
+
+if(CONFIG_ESP_WIFI_ENABLED AND CONFIG_ESP_WIFI_SOFTAP_SUPPORT)
+    list(APPEND srcs "src/scheme_softap.c")
+endif()
+
+if(CONFIG_BT_ENABLED)
+    if(CONFIG_BT_BLUEDROID_ENABLED OR CONFIG_BT_NIMBLE_ENABLED)
+        list(APPEND srcs
+            "src/scheme_ble.c")
+    endif()
+endif()
+
+idf_component_register(SRCS "${srcs}"
+                    INCLUDE_DIRS include
+                    PRIV_INCLUDE_DIRS src proto-c ${IDF_PATH}/components/protocomm/proto-c
```

Comment:
We had discussed that we should evaluate if the `protocomm` in itself can be pushed to the component manager first. Did you check that approach?

Reply:
@mahavirj The new `network_provisioning` should anyway work with the IDF v5.1 and v5.2, in which `protocomm` is a native component.

After the `network_provisioning` is pushed to component manager, we can do the same to `protocomm`, and remvoe both `wifi_provisioning` and `protocomm` from IDF.

Reply:
But will the code here still work once protocomm is pushed to the component registry?

Reply:
But if we push `protocomm` to component registry first, then publish `network_provisioning`, it will only work with latest IDF master.

How about this flow:
- push the new `network_provisioning` to component registry first, using the native `protocomm` in IDF
- push `protocomm` to component registry
- update `network_provisioning` version to use `protocomm` in component registry
- remove `protocomm` and `wifi_provisioning` from IDF

In this way, we will have `network_provisioning` versions work with IDF v5.1, v5.2 and later.

Reply:
By the way, why does protocomm component not add `proto-c/` subdirectory to public include paths, if this directory is required by other components? @mahavirj do you happen to know? I haven't noticed that we are adding `../protocomm/proto-c` to PRIV_INCLUDE_DIRS also in $IDF_PATH/components/wifi_provisioning/CMakeLists.txt.

Reply:
I don't see any specific reason, we can update the protocomm component to keep `proto-c` as public include path.

Reply:
@mahavirj I think we can firstly push this component to the component manager and discuss whether to move the `protocomm` component from idf component later. If the `protocomm` component is pushed to the component manager we can update this component to use the `protocomm` component from component manager